frame/bounties/src/lib.rs

```diff
@@ -544,17 +632,41 @@ decl_module! {
 				let bounty = maybe_bounty.take().ok_or(Error::<T>::InvalidIndex)?;
 				if let BountyStatus::PendingPayout { curator, beneficiary, unlock_at } = bounty.status {
 					ensure!(system::Module::<T>::block_number() >= unlock_at, Error::<T>::Premature);
+					// Get bounty account id
 					let bounty_account = Self::bounty_account_id(bounty_id);
 					let balance = T::Currency::free_balance(&bounty_account);
 					let fee = bounty.fee.min(balance); // just to be safe
-					let payout = balance.saturating_sub(fee);
+					// Make curator fee payment & unreserve the deposit
 					let _ = T::Currency::unreserve(&curator, bounty.curator_deposit);
 					let _ = T::Currency::transfer(&bounty_account, &curator, fee, AllowDeath); // should not fail
-					let _ = T::Currency::transfer(&bounty_account, &beneficiary, payout, AllowDeath); // should not fail
-					*maybe_bounty = None;
 
-					BountyDescriptions::remove(bounty_id);
+					// Make beneficiary payment
+					let payout = balance.saturating_sub(fee);
 
+					// TODO :: Have to recheck
+					// When master curator managing subbounties
+					// Master curator may have to close the bounty and transfer
```

Comment:
this doesn't make sense? if we want the fund go to treasury, curator should just `close_bounty`

Reply:
I hope then close_bounty() needs to be modified to support,

* Take signed origin ( Right now close_bounty() accepts only Root origin, ) & should be master curator.
*  if origin is from master curator, Need to support curator fee payment.